kotlin/lib/commonMain/kotlin/temperature/TemperaturCache.kt

```diff
@@ -0,0 +1,294 @@
+package temperature
+
+import hct.Hct
+import utils.ColorUtils.labFromArgb
+import utils.MathUtils
+import utils.MathUtils.sanitizeDegreesDouble
+import utils.MathUtils.sanitizeDegreesInt
+import kotlin.collections.ArrayList
+import kotlin.collections.HashMap
+import kotlin.math.*
+
+/**
+ * Design utilities using color temperature theory.
+ *
+ * <p>Analogous colors, complementary color, and cache to efficiently, lazily, generate data for
+ * calculations when needed.
+ */
+class TemperatureCache(private val input: Hct) {
```

Comment:
There is a typo in the file name, there is a missing `e` 🙂

Reply:
If this gets considered to merge, I can fix it.
However the filename doesn't really matter in Kotlin, so it's not urgent